content/learn/book/migration-guides/0.8-0.9/_index.md

```diff
@@ -0,0 +1,825 @@
+
+`add_plugins_with` has been replaced by `add_plugins` in combination with the builder pattern:
+
+```rust
+// Old (Bevy 0.8)
+app.add_plugins_with(DefaultPlugins, |group| group.disable::<AssetPlugin>());
+
+// New (Bevy 0.9)
+app.add_plugins(DefaultPlugins.build().disable::<AssetPlugin>());
+```
+
+### [Use plugin setup for resource only used at setup time](https://github.com/bevyengine/bevy/pull/6360)
+
+The `LogSettings` settings have been moved from a resource to `LogPlugin` configuration:
+
+```rust
+// Old (Bevy 0.8)
+app
+  .insert_resource(LogSettings {
+    level: Level::DEBUG,
+    filter: "wgpu=error,bevy_render=info,bevy_ecs=trace".to_string(),
+  })
+  .add_plugins(DefaultPlugins)
+
+// New (Bevy 0.9)
+app.add_plugins(DefaultPlugins.set(LogPlugin {
+    level: Level::DEBUG,
+    filter: "wgpu=error,bevy_render=info,bevy_ecs=trace".to_string(),
+}))
+```
+
+The `ImageSettings` settings have been moved from a resource to `ImagePlugin` configuration:
+
+```rust
+// Old (Bevy 0.8)
+app
+  .insert_resource(ImageSettings::default_nearest())
+  .add_plugins(DefaultPlugins)
+
+// New (Bevy 0.9)
+app.add_plugins(DefaultPlugins.set(ImagePlugin::default_nearest()))
+```
+
+The `DefaultTaskPoolOptions` settings have been moved from a resource to `CorePlugin::task_pool_options`:
+
+```rust
+// Old (Bevy 0.8)
+app
+  .insert_resource(DefaultTaskPoolOptions::with_num_threads(4))
+  .add_plugins(DefaultPlugins)
+
+// New (Bevy 0.9)
+app.add_plugins(DefaultPlugins.set(CorePlugin {
+  task_pool_options: TaskPoolOptions::with_num_threads(4),
+}))
+```
+
+### [Remove `AssetServer::watch_for_changes()`](https://github.com/bevyengine/bevy/pull/5968)
+
+`AssetServer::watch_for_changes()` was removed.
+Instead, set it directly on the `AssetPlugin`.
+
+```rust
+app
+  .add_plugin(DefaultPlugins.set(AssetPlugin {
+    watch_for_changes: true,
+    ..default()
+  }))
+```
+
+### [Spawn now takes a Bundle](https://github.com/bevyengine/bevy/pull/6054)
+
+```rust
+// Old (0.8):
+commands
+  .spawn()
+  .insert_bundle((A, B, C));
+// New (0.9)
+commands.spawn((A, B, C));
+
+// Old (0.8):
+commands.spawn_bundle((A, B, C));
+// New (0.9)
+commands.spawn((A, B, C));
+
+// Old (0.8):
+let entity = commands.spawn().id();
+// New (0.9)
+let entity = commands.spawn_empty().id();
+
+// Old (0.8)
+let entity = world.spawn().id();
+// New (0.9)
+let entity = world.spawn_empty();
+```
+
+### [Accept Bundles for insert and remove. Deprecate `insert`/`remove_bundle`](https://github.com/bevyengine/bevy/pull/6039)
+
+Replace `insert_bundle` with `insert`:
+
+```rust
+// Old (0.8)
+commands.spawn().insert_bundle(SomeBundle::default());
+// New (0.9)
+commands.spawn_empty().insert(SomeBundle::default());
+```
+
+Replace `remove_bundle` with `remove`:
+
+```rust
+// Old (0.8)
+commands.entity(some_entity).remove_bundle::<SomeBundle>();
+// New (0.9)
+commands.entity(some_entity).remove::<SomeBundle>();
+```
+
+Replace `remove_bundle_intersection` with `remove_intersection`:

```

Comment:
`remove_intersection`/`remove_bundle_intersection` is not a public API. This shouldn't be in the public facing migration notes.

Reply:
```suggestion

```

Reply:
@james7132 is this true? why is it a public method?

Reply:
Oh it seems like my earlier search didn't show any results because I searched the public docs for `bevy_ecs` for `remove_intersection`, not `remove_bundle_intersection`. Please revert the changes here. My mistake.

Reply:
@IceSentry cc